x-pack/plugins/ingest_manager/server/services/setup.ts

```diff
@@ -30,70 +30,93 @@ import { appContextService } from './app_context';
 const FLEET_ENROLL_USERNAME = 'fleet_enroll';
 const FLEET_ENROLL_ROLE = 'fleet_enroll';
 
+// the promise which tracks the setup
+let setupIngestStatus: Promise<void> | undefined;
```

Comment:
This is not going to work if we have multiple instances of Kibana no? Can we use a saved object, to keep the state of the setup?

Reply:
Yeah, I wondered about the same things. From Slack the other day:

> I initially had some logic for dealing with pending & succeeded state but they way we tried initially had issues in multi-kibana setups (and tests, iirc). We might need to move the state out of JS and into ES or somewhere else that's durable & shared by multiple Kibana

I did this because it's the simplest ~thing~ change that could possibly work, but I do think we'll want to move it to Saved Objects eventually.

Along those lines and https://github.com/elastic/kibana/issues/63123 perhaps we want a GET route to match `/fleet/setup`?

Do you think these need to be done in this PR or can they be discussed/implemented in a followup issue?

Reply:
I am in favor of doing it in this PR as I think this is probably not going to solve our problem

Reply:
Does the failure in multi-kibana exist in the code before #69505, e.g. in 7.8?

I'll see how quickly I can do a dead-simple boolean but, like I said in the description, I think we'll want more fine-grained permissions and I'd like to avoid designing that during the PR or writing such short-lived code if this works. I know, _"if"_ it works.